client/src/app/party-select/party-select.component.ts

```diff
@@ -44,13 +46,45 @@ export class PartySelectComponent implements OnInit {
   changeParty(index: number): void {
     this.hideUnimplementedJobs = false;
     this.jobPickerVisible = true;
-        this.jobPickerModalRef.afterClose.pipe(take(1)).subscribe(
-          (jobChosen: string) => {
-            if (jobChosen) {
-              this.dpsService.party[index] = jobChosen;
-              this.dpsService.updateAllStats();
-            }
-          });
+    this.selectableRoles = this.getAvailableRoles(
+        jobConsts.JOBS.get(this.dpsService.selfJob) || jobConsts.SCHOLAR,
+        index);
```

Comment:
nit: can this go on the previous line or will linter yell?

Reply:
Done.  Linter is sill a TODO. :x